### ISSUE TITLE
After NTP sent time is set to 1970.

### ISSUE THREAD

**Issue by Bolukan** [gmag11/NtpClient#79]
Hi, I use the developer branche with the ESP8266 example and with TimeZone added. I added this code in the loop:
```
  if (timeStatus() != timeNotSet) {
    if (now() != prevDisplay) { //update the display only if time has changed
      prevDisplay = now();
      printDateTime(CE, prevDisplay, "Amsterdam");
    }
  }
```
I get 1970 time readings, just around a NTP update I guess 50% of the time. Any idea? Second code snippet is with debug on.
```
19:42:41 Sun 20 Jan 2019 CET
19:42:42 Sun 20 Jan 2019 CET
19:42:43 Sun 20 Jan 2019 CET
01:00:00 Thu 01 Jan 1970 CET
Got NTP time: 18:42:43 20/01/2019
19:42:43 Sun 20 Jan 2019 CET
25 18:42:43 20/01/2019 Winter Time. WiFi is connected. Uptime:    0 days 00:02:10 since 18:40:37 20/01/2019
Free heap: 49256
19:42:44 Sun 20 Jan 2019 CET
19:42:45 Sun 20 Jan 2019 CET
19:42:46 Sun 20 Jan 2019 CET
19:42:47 Sun 20 Jan 2019 CET
```
and with debug on:
```
20:17:46 Sun 20 Jan 2019 CET  Amsterdam
20:17:47 Sun 20 Jan 2019 CET  Amsterdam
getTime
getTime - Resolving DNS of nl.pool.ntp.org
getTime - DNS result: -5
getTime - DNS Resolution in progress
20:17:48 Sun 20 Jan 2019 CET  Amsterdam
getIPClass - IPAddress: 94.228.220.14
dnsFound - 94.228.220.14
getTime
getTime - DNS name IP solved: 94.228.220.14
getTime - Starting UDP. IP: 94.228.220.14
getTime - Sending UDP packet

E3 00 06 EC | 00 00 00 00 | 00 00 00 00 | 31 4E 31 34
00 00 00 00 | 00 00 00 00 | 00 00 00 00 | 00 00 00 00
00 00 00 00 | 00 00 00 00 | 00 00 00 00 | 00 00 00 00

UDP packet sent
getTime - NTP request sent
01:00:00 Thu 01 Jan 1970 CET  Amsterdam
Secs: 3757000668
No daylight
Sync frequency set low
Successful NTP sync at 19:17:48 20/01/2019
UDP packet received
UDP Packet Type: Unicast, From: 94.228.220.14:123, To: 192.168.1.182:49156, Length: 48, Data:

24 02 06 E9 | 00 00 00 A2 | 00 00 05 B1 | C1 4F ED 0E
DF EF 46 40 | 1F 90 00 88 | 00 00 00 00 | 00 00 00 00
DF EF 47 DC | FF 40 4B 2C | DF EF 47 DC | FF 43 46 23

Got NTP time: 19:17:48 20/01/2019
20:17:48 Sun 20 Jan 2019 CET  Amsterdam
20:17:49 Sun 20 Jan 2019 CET  Amsterdam
```
PS: millis() is not effected.

**Comment by Bolukan** [comment]
Guess I found it: getTime returns 0, which is used in setTime.
https://github.com/gmag11/NtpClient/blob/a86b769037b42b8ca471250944394ab6c52afc49/src/NTPClientLib.cpp#L285
It is not the only place 0 is returned. Maybe reroute it to a local setTime function that only forwards the command if it is not zero?

**Comment by gmag11** [comment]
Hi. Setting time to 0 should be ignored by Time library. A workaround would be change some 0s by `now()`

**Comment by Bolukan** [comment]
The last part I don't understand, because the 'now' of my device is for a short period reset to 1970. Can you explain what you imagine? To freeze time at the main application?

I offer you to write a pull request for NtpClient to not forward 0's to TimeLib, if you are open to consider such a solution. But I understand you have the opinion TimeLib should show other behaviour.

**Comment by Bolukan** [comment]
https://github.com/PaulStoffregen/Time/issues/121

**Comment by bill-orange** [comment]
I added this:
```
while ((year(now()) < 1980) || (year(now()) > 2035)) {
    int i;
    delay (500);
    Serial.print("+");
    NTP.begin("0.us.pool.ntp.org", TIMEZONE, true);
    i++;
    if (i > 10) {
      ESP.restart();
    }
  }
```

**Comment by bill-orange** [comment]
Rats. I am still occasionally getting a ‘1970’ after a gettime().

**Comment by gmag11** [comment]
Hi, can you reproduce this on latest code?

**Comment by Bolukan** [comment]
No problems anymore with temporary 1970 flashbacks :-)
```
NTP Time Timezone Example
Sync interval set to 63
Set NTP timeout to 1500 ms
Connected to SSID XXXXXX @ bssid xx:xx:xx:xx:xx:xx channel 1
Got IP: 192.168.1.xx mask 255.255.255.0 gateway 192.168.1.1
NTP server set to nl.pool.ntp.org
NTP time zone set to: 0
--Set daylight saving OFF
getTime - NTP request sent
dnsFound - Get time
1970-01-01 01:00:05 NTP request sent
Secs: 3778942216
No daylight
Sync frequency set low
Successful NTP sync at 18:10:16 01/10/2019
UDP packet received
UDP Packet Type: Unicast, From: 84.245.9.254:123, To: 192.168.1.xx:55148, Length: 48, Data:

24 01 06 ED | 00 00 00 01 | 00 00 00 01 | 50 50 53 30
E1 3E 14 FC | 9E 80 25 54 | 00 00 00 00 | 00 00 00 00
E1 3E 15 08 | 27 1B D7 FB | E1 3E 15 08 | 27 29 AC 45

2019-10-01 20:10:16 <- Got new NTP time

<snip>

getTime - Resolving DNS of nl.pool.ntp.org
getTime - DNS result: 0
getIPClass - IPAddress: 84.245.9.254
getTime - DNS name IP solved: 84.245.9.254
getTime - Starting UDP. IP: 84.245.9.254
getTime - Sending UDP packet

E3 00 06 EC | 00 00 00 00 | 00 00 00 00 | 31 4E 31 34 
00 00 00 00 | 00 00 00 00 | 00 00 00 00 | 00 00 00 00 
00 00 00 00 | 00 00 00 00 | 00 00 00 00 | 00 00 00 00 

UDP packet sent
getTime - NTP request sent
18:11:19 01/10/2019
2019-10-01 20:11:19 NTP request sent
Secs: 3778942279
No daylight
Sync frequency set low
Successful NTP sync at 18:11:19 01/10/2019
UDP packet received
UDP Packet Type: Unicast, From: 84.245.9.254:123, To: 192.168.1.xx:55149, Length: 48, Data:

24 01 06 ED | 00 00 00 01 | 00 00 00 01 | 50 50 53 30
E1 3E 15 3B | 59 7E 12 E6 | 00 00 00 00 | 00 00 00 00
E1 3E 15 47 | 31 93 44 2C | E1 3E 15 47 | 31 9E C5 45

2019-10-01 20:11:19 <- Got new NTP time
```

**Comment by gmag11** [comment]
Great! I close the issue. Thank you for your help
